url-updater/src/main/java/com/devonfw/tools/ide/url/folderhandling/UrlChecksum.java

```diff
@@ -0,0 +1,129 @@
+package com.devonfw.tools.ide.url.folderhandling;
+
+import com.devonfw.tools.ide.url.folderhandling.abstractUrlClasses.AbstractUrlFile;
+
+import java.io.*;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class UrlChecksum extends AbstractUrlFile {
+
+    final HttpClient client = HttpClient.newBuilder().build();
+    final String hashAlgorithm = "SHA-256";
+    private String checksum;
+
+    /**
+     * The constructor.
+     *
+     * @param parent the {@link #getParent() parent folder}.
+     * @param name   the {@link #getName() filename}.
+     */
+    public UrlChecksum(UrlVersion parent, String name) {
+        super(parent, name + ".sha256");
+    }
+
+    public String getChecksum() {
+        return checksum;
+    }
+
+    public void setChecksum(String checksum) {
+        this.checksum = checksum;
+    }
+
+    /**
+     * @param url the url of the download file
+     * @return the input stream of requested url
+     */
+    private InputStream doGetResponseInputStream(String url) {
+        try {
+            HttpRequest request1 = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .GET()
+                    .build();
+            return client.send(request1, HttpResponse.BodyHandlers.ofInputStream()).body();
+        } catch (IOException | InterruptedException exception) {
+            throw new IllegalStateException("Failed to retrieve response body from url: " + url, exception);
+        } catch (IllegalArgumentException e) {
+            System.out.println("Error while getting response body from url {}\", url, e");
```

Comment:
Use Logger and ensure the stacktrace does not get lost

Reply:
done